internal/provider/common.go

```diff
@@ -26,10 +26,20 @@ type NodeClientCollection interface {
 	getNodeClient(sub *substrate.Substrate, nodeID uint32) (*client.NodeClient, error)
 }
 
+type Progress struct {
+	time    time.Time
+	stateOk int
+}
+
 func waitDeployment(ctx context.Context, nodeClient *client.NodeClient, deploymentID uint64, version uint32) error {
 	done := false
-	for start := time.Now(); time.Since(start) < 4*time.Minute; time.Sleep(1 * time.Second) {
+	last_progress := Progress{time.Now(), 0}
+	is_progressing := true
+
+	for start := time.Now(); time.Since(start) < 4*time.Minute || is_progressing; time.Sleep(1 * time.Second) {
```

Comment:
It was done like this in the old days but this results in ~60 requests per minute to just check for the deployment. And `sleep` has another bad effect that it doesn't respect the context, so if a timeout was set or the context was cancelled, the sleep won't stop except after the sleeping amount is done. I suggest using something like [this](https://pkg.go.dev/github.com/cenkalti/backoff/v3#ExponentialBackOff) to make fast deployment return quickly and not cause too much load on the node. A sensible paramater list for the exponential back off object might be:
- InitialInterval: 3 seconds
- Multiplier: 1.25
- MaxInterval: 40 seconds
- DefaultMaxElapsedTime: large enough to not fail early for long deployments (50 minutes?)